src/electron/views/device-connect-view/components/device-connect-view-container.tsx

```diff
@@ -26,34 +31,48 @@ export type DeviceConnectViewContainerProps = {
     userConfigurationStoreData: UserConfigurationStoreData;
     deviceStoreData: DeviceStoreData;
     windowStateStoreData: WindowStateStoreData;
+    featureFlagStoreData: FeatureFlagStoreData;
+    androidSetupStoreData: AndroidSetupStoreData;
 };
 
 export const DeviceConnectViewContainer = NamedFC<DeviceConnectViewContainerProps>(
     'DeviceConnectViewContainer',
     props => {
         return (
-            <div className={deviceConnectView}>
-                <WindowTitle
-                    pageTitle={'Connect to your Android device'}
-                    deps={props.deps}
-                    windowStateStoreData={props.windowStateStoreData}
-                >
-                    <HeaderIcon invertColors deps={props.deps} />
-                </WindowTitle>
-                <div className={mainContentWrapper}>
-                    <DeviceConnectBody
-                        deps={props.deps}
-                        viewState={{
-                            deviceConnectState: props.deviceStoreData.deviceConnectState,
-                            connectedDevice: props.deviceStoreData.connectedDevice,
-                        }}
-                    />
-                    <TelemetryPermissionDialog
-                        deps={props.deps}
-                        isFirstTime={props.userConfigurationStoreData.isFirstTime}
-                    />
-                </div>
-            </div>
+            <FlaggedComponent
+                featureFlagStoreData={props.featureFlagStoreData}
+                deps={props.deps}
+                featureFlag={UnifiedFeatureFlags.adbSetupView}
+                enableJSXElement={<AndroidSetupStepComponent {...props} />}
+                disableJSXElement={productionDeviceConnectView(props)}
+            ></FlaggedComponent>
         );
     },
 );
+
+const productionDeviceConnectView = (props: DeviceConnectViewContainerProps) => {
+    return (
+        <div className={deviceConnectView}>
+            <WindowTitle
+                pageTitle={'Connect to your Android device'}
+                deps={props.deps}
+                windowStateStoreData={props.windowStateStoreData}
+            >
+                <HeaderIcon invertColors deps={props.deps} />
+            </WindowTitle>
+            <div className={mainContentWrapper}>
+                <DeviceConnectBody
+                    deps={props.deps}
+                    viewState={{
+                        deviceConnectState: props.deviceStoreData.deviceConnectState,
+                        connectedDevice: props.deviceStoreData.connectedDevice,
+                    }}
+                />
+                <TelemetryPermissionDialog
```

Comment:
Should the telemetry dialog be part of one of the setup step tsx components?

Reply:
I don't think so. Imagine a future version where the user needs to explicitly choose and android scanning option--we'd probably still want the telemetry dialog to be the first thing to appear

Reply:
+1. I would lean towards just wrapping the `DeviceConnectBody` in the `FlaggedComponent` and leaving the `WindowTitle` and `TelemetryPermissionDialog` as-is regardless of the flag